code-challenges.test.js

```diff
@@ -14,54 +14,182 @@
+--------- INPUT / OUTPUT:
+input 1: object w/number values
+output 1: array w/number values sorted from least to greatest.
+
+--------- STEPS/METHODS:
+1. Declare a temporary variable (newArr), which we will be passing the values from our object into.
+2. Use the method Object.values() to obtain all values from the object being passed in.
+3. Use the .sort method with an argument of ((a, b) => a - b) in order to sort the items in the array from least to greatest.
+4. Return newArr.
+*/
+
+const objSort = (object) => {
+  // Filling the empty arr with the values of object, then sorting them.
+  newArr = Object.values(object).sort((a, b) => a - b);
+  // Returning the sorted array.
+  return newArr;
+};
+
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const accountTransactions1 = [100, -17, -23, -9]
-// Expected output: [100, 83, 60, 51]
+describe("arrSummation", () => {
+  it(`Takes in an array and returns an array of the accumulating sum.`, () => {
+    const accountTransactions1 = [100, -17, -23, -9];
+    const accountTransactions2 = [250, -89, 100, -96];
+    const accountTransactions3 = [];
 
-const accountTransactions2 = [250, -89, 100, -96]
-// Expected output: [250, 161, 261, 165]
+    expect(arrSummation(accountTransactions1)).toEqual([100, 83, 60, 51]);
+    expect(arrSummation(accountTransactions2)).toEqual([250, 161, 261, 165]);
+    expect(arrSummation(accountTransactions3)).toEqual([]);
+  });
+});
 
-const accountTransactions3 = []
-// Expected output: []
+// GOOD FAIL: arrSummation is not defined
 
 // b) Create the function that makes the test pass.
+/*
+--------- FUNCTION INFORMATION:
+Describe what the function will do: Function will take in an array and return an array of the accumulating sum.
+Function Name: arrSummation
+Function Parameter(s): 1, array of numbers
+--------- INPUT / OUTPUT:
+input 1: arr of numbers
+output 1: arr of same length, with each element returning the sum of adding it to the previous element.
+
+input 2: empty array
+output 2: []
+--------- STEPS/METHODS:
+1. Declare a variable to hold our starting number, which will be set to 0.
+2. Use the ternary operator to check if the array.length is greater than 0, so that it will return an empty array if the one passed in has no value.
+3. Since I want to do something to EVERY element in the array AND return an array of the same length, I'll use .map() to iterate.
+4. Add the value of the current iteration to the value stored in the temporary variable, reset the value of the temporary variable with the current value, then return the value to the array.
+*/
+
+// ----- VISUAL BREAKDOWN -----
+// Expect: [100, 83, 60, 51]
+// 0. 100
+// 1. 100 - 17 = 83
+// 2. 83 - 23 = 60
+// 3. 60 - 9 = 51
+
+const arrSummation = (arr) => {
```

Comment:
Excellent work!  Really happy to see that you utilized higher-order functions on this problem!  👏